OnBaseDocsApi/Controllers/OnBaseDocsController.cs

```diff
@@ -57,7 +61,200 @@ public IHttpActionResult GetFile(int id)
             });
         }
 
-        IHttpActionResult TryHandleDocRequest(int id, Func<Application, Document, IHttpActionResult> handler)
+        [HttpPost]
+        [ActionName("")]
+        public async Task<IHttpActionResult> Post()
+        {
+            // We only support multi-part post content.
+            if (!Request.Content.IsMimeMultipartContent())
+            {
+                return BadRequestResult("Uploading a document requires a multi-part POST request.");
+            }
+
+            Stream docStream = null;
+            string docExt = null;
+            DocumentPostAttributes docAttr = null;
+
+            // We bundle the bad requests in one response.
+            var errors = new List<Error>();
+
+            try
+            {
+                var provider = await Request.Content.ReadAsMultipartAsync(
+                    new MultipartMemoryStreamProvider());
+
+                foreach (HttpContent content in provider.Contents)
+                {
+                    // The Content-Disposition header is required.
+                    if (content.Headers.ContentDisposition == null)
+                        return BadRequestResult($"A Content-Disposition header is required.");
```

Comment:
Same, I think we can use normal string here.

Reply:
Agree.